toolbox/fdc3-for-web/fdc3-web-impl/src/handlers/BroadcastHandler.ts

```diff
@@ -68,11 +71,38 @@ export class BroadcastHandler implements MessageHandler {
 
   shutdown(): void {}
 
-  getCurrentChannel(from: AppIdentifier): ChannelState | null {
-    // instanceId is optional
-    if (!from.instanceId) {
-      return null;
-    }
+  cleanup(instanceId: InstanceID, sc: ServerContext<AppRegistration>): void {
```

Comment:
we need to test this

Reply:
Yes, cleanup was not being performed prior to this (when reconnect was fixed, it generated an additional copy of each broadcast or intent event for each reconnection). I guess that's a case of checking outgoing messages are correct after a reconnect and over multiple broadcasts (the last message will be correct, but need to check it wasn't duplicated - so need to look back through more messages)?

Reply:
You'll need to make some changes to `TestServerContext` for that - at the moment, it won't send messages to destinations that have disconnected.  But if you changed it so that it sent to some dead-letter address you could add those messages to the `postedMessages` list and check them in the test.

Reply:
The important thing is that it has disconnected then reconnected, but state got cleaned up in between. Hence the challenge is simulating the disconnection - I think just calling the cleanup function on the server for the UUID, then registering again _should_ do that....

Reply:
It does, have manually broken it to make sure the test will catch a lack of cleanup and it appears to.

Reply:
Added testing